### PR TITLE
feat(ui): 531 file management

### LINKED ISSUE #531: feat(ui) File Management Page - UI components
### User Story

**As a** LFAI user
**I want** upload files to LFAI
**So that** so I can interact with the files

### Acceptance Criteria

**Given** I am using LFAI `Design: LFAI`
**When** I click on the Settings icon in the header
**Then** I see an option entitled "File Management" `Design: Settings Right Panel Open`

**When** I click on the option "File Management"
**Then** I am navigated to the "File Management" page `Design: File Management Page`
**And** I see a breadcrumb "Chat / File Management"
**And** I see a heading "File Management" (typography: heading-05)
**And** I see a scrollable data table a search field
**And** an "Upload" button
**And** 2 columns "Name" and "Date Added"

**When** I type into the search field
**Then** the data table to filtered only to include results that contain the typed text in the files

**When** I click on the "Name" header
**Then** the data table sorts by the "Name" column

**When** I click on the "Date Added" header
**Then** the data table sorts by the "Date Added" column

###Note: The functionality of the "Import" button and row hover will come in a subsequent story 

### Additional context
Designs:
- [LFAI](https://www.figma.com/design/Vaw2TCF9DPAp1msj5TWkl0/Leapfrog-AI?m=dev&node-id=2675%3A45857&t=9cmnlkvPJLpancmy-1)
- [Settings Right Panel Open](https://www.figma.com/design/Vaw2TCF9DPAp1msj5TWkl0/Leapfrog-AI?m=dev&node-id=2687%3A169631&t=9cmnlkvPJLpancmy-1)
- [File Management Page](https://www.figma.com/design/Vaw2TCF9DPAp1msj5TWkl0/Leapfrog-AI?m=dev&node-id=2816%3A129691&t=9cmnlkvPJLpancmy-1)

### LINKED ISSUE #540: bug(ui) Tagline being cutoff on assistant tile
### Environment
Device and OS: MacBook Pro 
App/package versions: macOS Sonoma 14.4.1 (23E224)
Kubernetes distro being used: N/A
Other:

### Steps to reproduce
1. Create an assistant with 1 very long continuous string in the tagline field (i.e. reallyreallyreallyreallyreallylooooooooooooooooong)
2. Save assistant
3. See tagline cutoff on assistant tile

### Expected result
Text should wrap to 2nd round and then truncate as to not go past 2 lines

### Actual Result
Text gets cutoff on 1st line

### Visual Proof (screenshots, videos, text, etc)

Image

### LINKED ISSUE #550: feat(ui) File Management Functionality
### User Story

**As a** LFAI user
**I want** upload files to LFAI
**So that** so I can interact with the files

### Acceptance Criteria

**Given** I am on the File Management page `Design: File Management Page`
**When** I click the upload button
**Then** a file explorer window opens `Design: File Management File Explorer`
**And** I am able to select .pdf or .txt formatted files (if we can support multi select then that is preferred, but no big deal if we cannot)

**When** I select a supported file type and upload it to LFAI
**Then** a new row item appears in the data table with a loading component to the left of the "Name" `Design: File Management Uploading`
**And** the name appears in the same typography as the rest of the data table expect the font color is text-placeholder
**And** nothing appears in the "Date Added" column

**When** a file is being uploaded to the data table
**Then** the loading component's state is in progress while the file is being uploaded `Design: File Management Uploading`
**And When** the file upload completes successfully
**Then** the loading component's state changes to complete `Design: File Management File Upload Success`
**And** the text color changes to text-secondary
**And** the date is added to the "Date Added" column
**And** a "Imported Successfully" toast message appears
**But When** the file upload fails
**Then** the loading component's state changes to failed `Design: File Management File Upload Fail`
**And** the text color remains text-placeholder
**And** still nothing appears in the "Date Added" column
**And** a "Import Failed" toast message appears

**When** the loading component's state changes to complete
**Then** the complete icon appears for 1.5 seconds `Design: File Management File Upload Success`
**And Then** disappears `Design: File Management File Added`
**But When** the loading component's state changes to failed
**Then** the fail icon appears for 1.5 seconds `Design: File Management File Upload Fail`
**And Then** the entire row item disappears `Design: File Management File Added`

### Additional context
Designs:
- [File Management Page](https://www.figma.com/design/Vaw2TCF9DPAp1msj5TWkl0/Leapfrog-AI?m=dev&node-id=2816%3A129691&t=9cmnlkvPJLpancmy-1)
- [File Management File Explorer](https://www.figma.com/design/Vaw2TCF9DPAp1msj5TWkl0/Leapfrog-AI?m=dev&node-id=3303%3A233253&t=9cmnlkvPJLpancmy-1)
- [File Management Uploading](https://www.figma.com/design/Vaw2TCF9DPAp1msj5TWkl0/Leapfrog-AI?m=dev&node-id=3336%3A247238&t=9cmnlkvPJLpancmy-1)
- [File Management File Upload Success](https://www.figma.com/design/Vaw2TCF9DPAp1msj5TWkl0/Leapfrog-AI?m=dev&node-id=3336%3A248566&t=9cmnlkvPJLpancmy-1)
- [File Management File Upload Fail](https://www.figma.com/design/Vaw2TCF9DPAp1msj5TWkl0/Leapfrog-AI?m=dev&node-id=3336%3A249633&t=9cmnlkvPJLpancmy-1)
- [File Management File Added](https://www.figma.com/design/Vaw2TCF9DPAp1msj5TWkl0/Leapfrog-AI?m=dev&node-id=3303%3A234296&t=9cmnlkvPJLpancmy-1)

### PR DESCRIPTION
**Pull request by andrewrisse**
closes #531 
closes #550 
closes #540

Adds a file management page accessible through the top right settings icon.

Allows multiple file upload and delete.

Show upload success and error icon animations that disappear after 1.5 seconds. Also has toasts.

Allows search by filename and created_at date.

Allows sort by filename or created_at fields.

**UPDATE**
This PR also now includes a refactor to change the data fetching and invalidation pattern for assistants and files.

The code to fetch assistants and files was moved to an API endpoint so that +page.ts files could call those endpoints using the provided (not native) fetch function [Docs Ref](https://kit.svelte.dev/docs/load#making-fetch-requests). 

This allows us to call invalidate('/api/some_route') instead of invalidateAll, resulting in less network traffic and redundant data fetching.

The fetching for threads was also moved to a parent +layout.server.ts so the threads weren't re-fetched everytime the /chat route was visited.

Additionally, the Supabase docs describing how to set up auth with Sveltekit were updated and include a new 'safeGetSession' function. This PR also updates the hooks.server.ts file and associated server side locals to include the new function.

PR is deployed on leapfrog-04 at ai.uds.dev

### CODE REVIEW

**Review by gregclark-defenseunicorns — APPROVED** [review]
Design review passed

**Comment by andrewrisse** [comment]
[file-management.webm](https://github.com/defenseunicorns/leapfrogai/assets/52644157/b69a5c5d-8bac-402e-a787-31e750873116)

File: src/leapfrogai_ui/src/routes/chat/(settings)/file-management/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    submit(); //upload all files
+  };
+
+  $: if (uploadedFiles.length > 0) {
+    //files selected by user
+    handleUpload();
+  }
+</script>
+
+<div class="file-management-container">
```
Comment (BillyFigueroa) [inline src/leapfrogai_ui/src/routes/chat/(settings)/file-management/+page.svelte:144]: This is a very large file. Is there a way to grab some of this markup and create a component?

File: src/leapfrogai_ui/testUtils/fakeData/index.ts [hunk 2]
```diff
@@ -3,17 +3,18 @@ import { faker } from '@faker-js/faker';
 import { assistantDefaults, DEFAULT_ASSISTANT_TEMP } from '../../src/lib/constants';
 import type { AssistantInput, LFAssistant } from '../../src/lib/types/assistants';
 import type { LFMessage } from '../../src/lib/types/messages';
-import type { LFThread, Roles } from '../../src/lib/types/threads';
+import type { LFThread } from '../../src/lib/types/threads';
```
Comment (BillyFigueroa) [inline src/leapfrogai_ui/testUtils/fakeData/index.ts:6]: Are these some of the paths that cannot be updated?

**Review by BillyFigueroa — APPROVED** [review]

File: src/leapfrogai_ui/testUtils/fakeData/index.ts (hunk 2 above)
Comment (andrewrisse) [reply src/leapfrogai_ui/testUtils/fakeData/index.ts:6]: Ya, the fake data helpers are used in Playwright and I haven't had time yet to figure out how to get the relative imports to work. I may just have to move the playwright tests into the src directory

File: src/leapfrogai_ui/src/routes/chat/(settings)/file-management/+page.svelte (hunk 1 above)
Comment (andrewrisse) [reply src/leapfrogai_ui/src/routes/chat/(settings)/file-management/+page.svelte:144]: I'm actually not sure how to break this one up, the form needs all these functions, and the markup is just a little big for the custom icon animation work we have going on

**Review by BillyFigueroa — APPROVED** [review]

**Review by CollectiveUnicorn — APPROVED** [review]

**Review by gregclark-defenseunicorns — APPROVED** [review]
Design pass

**Pull request merged**